backend/pvm.pm

```diff
@@ -96,30 +97,38 @@ sub pvmctl {
     my $lpar = $vars->{LPAR};
 
     if ($type =~ /lpar/) {
-        push(@cmd, "-i", "name=" . $lpar) if ($action =~ /power|restart|delete/);
+        gen_params @cmd, "i", "name=$lpar" if ($lpar && $action =~ /power|restart|delete/);
         if ($action =~ /create/) {
             my ($type, $action, $cpu, $memory) = @_;
-            push(@cmd, qw(--proc-type shared --sharing-mode uncapped --type AIX/Linux));
-            push(@cmd, "--proc", $cpu, "--mem", $memory, "--name", $lpar);
-            push(@cmd, "--proc-unit", $cpu * 0.05);
+            dd_gen_params @cmd, "proc-type",    "shared";
+            dd_gen_params @cmd, "sharing-mode", "uncapped";
+            dd_gen_params @cmd, "type",         "AIX/Linux";
+            dd_gen_params @cmd, "proc",         $cpu if $cpu;
+            dd_gen_params @cmd, "mem",          $memory if $memory;
+            dd_gen_params @cmd, "name",         $lpar if $lpar;
+            dd_gen_params @cmd, "proc-unit",    $cpu * 0.05 if $cpu;
         }
     }
     elsif ($type =~ /scsi/) {
         my ($type, $action, $kind, $file, $target) = @_;
         #so far only disks are reatachable to the master LPAR
         #set it back to default if argument is omited
         $lpar = $target if $target;
-        push(@cmd, "--type", $kind, "--stor-id", $file);
-        push(@cmd, "--lpar", "name=" . $lpar);
-        push(@cmd, "--vg", "name=rootvg") if ($type =~ /lv/);
+        dd_gen_params @cmd, "type",    $kind         if $kind;
```

Comment:
all these post ifs are very strange. Because the command will not work if one or the other isn't set